### DISCUSSION TITLE
updating nested map specific value reset the entire map?

### DISCUSSION THREAD

**Discussion by Mahsanali0786** [cachapa/firedart#108] [Q&A]
await Firestore.instance
                  .collection('myCollection')
                  .document('myDocument')
                  .update(
                {'map.map2.field': 'value'},
              );
in the above code when i try to update the field in map it  just delete all the values. Kindly guide me?

**Comment by cachapa** [comment]
I'm not sure exactly what your data is supposed to look like, but it looks like you're trying to access nested maps inside `update()`.

I don't think Firebase can interpret 'map.map2.field' as "field inside two nested maps", but rather it's overwriting the field 'map.map2.field' with 'value'.

Your options are to either get the latest version of the map, update it locally and write it again, or use nested document/collections.
